store/statstore/stat_store.go

```diff
@@ -327,10 +337,26 @@ func (ss *StatStore) Get(start, end time.Time) []TimePoint {
 
 	var result []TimePoint
 
-	if ss.IsEmpty() || (start.After(end) && end.After(time.Time{})) {
+	if start.After(end) && end.After(time.Time{}) {
+		return result
+	}
+
+	// Generate a TimePoint for the lastPut, if within range
+	low := start.Equal(time.Time{}) || start.Before(ss.lastPut.stamp)
+	hi := end.Equal(time.Time{}) || !end.Before(ss.lastPut.stamp)
+	if ss.lastPut.stamp.After(time.Time{}) && low && hi {
```

Comment:
we are actually maintaining a count within lastPut to be able to calculate the cumulative average. I will change the implementation to use that counter instead.